src/export.ts

```diff
@@ -70,7 +70,7 @@ export const backupFromDoc = async <T>(
             if (data[collectionName][doc.id][refKey]) {
               if (Array.isArray(data[collectionName][doc.id][refKey])) {
                 for (let val of data[collectionName][doc.id][refKey]) {
```

Comment:
Thanks, do you think that we should also remove the forloop `for (let val of data[collectionName][doc.id][refKey])`?

Did it work when you test it?

Reply:
I've "tried" basically by updating transpiled code in `dist` folder of your library (i know, its bad, thus the PR ^^).

Before doing this it would not let me export arrays containing references, but then once I've added this line and re-run my `yarn firestore:sync export` script (see attached) it started to behave as it should.

Anyway, I think we have two choices here (or maybe more):

```ts
if (Array.isArray(data[collectionName][doc.id][refKey])) {
  for (let val of data[collectionName][doc.id][refKey]) {
    data[collectionName][doc.id][refKey] = db.doc(getPath(val));
  }
}
```

```ts
if (Array.isArray(data[collectionName][doc.id][refKey])) {
  data[collectionName][doc.id][refKey] = data[collectionName][doc.id][refKey]
    .map((ref) => db.doc(getPath(ref)));
}
```

What do you think is better? Also could you please validate that it works for you, as I'm pretty newbie with Firebase, i wouldn't want to mess with people's projects ^^

Reply:
@WNemencha can you send a snapshot of your sample data in firestore & the result after you backup?